src/StatsdClient/Bufferize/StatsBufferize.cs

```diff
@@ -62,15 +62,15 @@ public void OnNewValue(Stats stats)
                 using (stats)
                 {
                     _statsRouter.Route(stats);
-                    _stopwatch = null;
+                    _stopwatch.Reset();
```

Comment:
I use `_stopwatch = null;` and not `_stopwatch.Reset();` because I would like to avoid calling `_stopwatch.Reset();` in the hot path of the application. I did  some tests a year ago that show using `_stopwatch = null;` on an heavy load has a better performance compared to `_stopwatch.Reset();`. 
I have just run a benchmark with `BenchmarkDotNet` that have shown similar results.

Anyway, I think there is an easy solution that performs a single allocation of `Stopwatch` and do not call `_stopwatch.Reset();` in the hot path.
A `bool resetTimer` can be set at `true` in `OnNewValue`. This boolean can be used inside `OnIdle()` to properly reset the timer. What do you think?

Please let me know if you want I made this change.

Reply:
I'm curious about your benchmark. `Reset` is just zeroing a few fields, it shouldn't be costly: https://referencesource.microsoft.com/#System/services/monitoring/system/diagnosticts/Stopwatch.cs,103

Are you sure you're not calling `Restart` instead?

Reply:
I have also noticed that `Reset` does not do a lot of things. The improvement  of `_stopwatch = null;` versus `_stopwatch.Reset();` happens when `OnIdle` is called rarely.

In general, I prefer to not make any assumptions about the implementation when there is an easy alternative solution.

Anyway, feel free to resolve this conversion. I am going to profile the whole code later so I can update this code if I spot a performance bottleneck.

Reply:
Done in https://github.com/DataDog/dogstatsd-csharp-client/pull/142/commits/7e1c28a5771c2bbe2dceee81708997d67aa71a47

Reply:
Actually, https://github.com/DataDog/dogstatsd-csharp-client/pull/142/commits/e6ce37ae0f5a74d5c7921eba3ee43ef5963028f3 😛 
I don't know why I assumed that the two methods would be called from different threads.